org.adempiere.base/src/org/compiere/util/Msg.java

```diff
@@ -143,30 +144,29 @@ private Msg()
 		ResultSet rs = null;
 		try
 		{
+			
 			if (AD_Language == null || AD_Language.length() == 0 || Env.isBaseLanguage(AD_Language, "AD_Language"))
 				pstmt = DB.prepareStatement("SELECT Value, MsgText, MsgTip FROM AD_Message",  null);
 			else
 			{
 				pstmt = DB.prepareStatement("SELECT m.Value, t.MsgText, t.MsgTip "
 					+ "FROM AD_Message_Trl t, AD_Message m "
 					+ "WHERE m.AD_Message_ID=t.AD_Message_ID"
+					+ " AND t.AD_Client_ID = 0" // load only translated messages at System level (using Value as key)
 					+ " AND t.AD_Language=?", null);
 				pstmt.setString(1, AD_Language);
 			}
-			rs = pstmt.executeQuery();
 
-			//	get values
-			while (rs.next())
-			{
-				String AD_Message = rs.getString(1);
-				StringBuilder MsgText = new StringBuilder();
-				MsgText.append(rs.getString(2));
-				String MsgTip = rs.getString(3);
-				//
-				if (MsgTip != null)			//	messageTip on next line, if exists
-					MsgText.append(" ").append(SEPARATOR).append(MsgTip);
-				msg.put(AD_Message, MsgText.toString());
-			}
+			addMessagesInCache(pstmt, rs, msg);
+
+			// load translated messages at tenant level (using AD_Client_ID|Value as key)
+			pstmt = DB.prepareStatement("SELECT t.AD_Client_ID || '|' || m.Value, t.MsgText, t.MsgTip"
+					+ " FROM AD_Message_Trl t, AD_Message m"
+					+ " WHERE m.AD_Message_ID=t.AD_Message_ID"
+					+ " AND t.AD_Client_ID != 0"
+					+ " AND t.AD_Language=?", null);
```

Comment:
missing isactive check

Reply:
Should have t.IsActive='Y' too

Reply:
hum, weird, i added the 3 missing checks (one per query) ; can you recheck ?

Reply:
you added IsActive check for AD_Message but not AD_Message_Trl

Reply:
ok done